### PR TITLE
Feature/deploy proxy

### PR DESCRIPTION
**Pull request by cducrest**
based on https://github.com/trustlines-protocol/contracts/pull/248
Only one commit to review

### CODE REVIEW

File: py-deploy/tldeploy/identity.py [hunk 1]
```diff
@@ -267,3 +271,55 @@ def filled_and_signed_meta_transaction(
...
+        abi=factory_interface["abi"],
+        bytecode=factory_interface["bytecode"],
+    )
+
+    tx_id = factory.functions.deployProxy(
+        initcode, implementation_address, signature
+    ).transact({"from": web3.eth.accounts[0]})
+    wait_for_successful_transaction_receipt(web3, tx_id)
+
+    proxy_address = build_create2_address(factory_address, initcode)
```
Comment (berndbohmeier) [inline py-deploy/tldeploy/identity.py]: Can we not better ask for the address from the deployment transaction. We have events for it right? This way it is easier to spot an issue if there is any I think

File: py-deploy/tldeploy/identity.py [hunk 2]
```diff
@@ -267,3 +271,55 @@ def filled_and_signed_meta_transaction(
...
+    proxied_identity = web3.eth.contract(
+        address=proxy_address,
+        abi=identity_interface["abi"],
+        bytecode=identity_interface["bytecode"],
+    )
+    return proxied_identity
+
+
+def recover_proxy_deployment_signature_owner(
+    web3, facotry_address, implementation_address, signature
```
Comment (berndbohmeier) [inline py-deploy/tldeploy/identity.py]: spelling: factory

File: py-deploy/tldeploy/identity.py (hunk 1 above)
Comment (cducrest) [reply py-deploy/tldeploy/identity.py]: Yes, we can use the events

**Comment by berndbohmeier** [comment]
All commits are in #248

**Pull request closed without merging**